Sources/SwiftDocC/Semantics/Metadata/PageKind.swift

```diff
@@ -0,0 +1,76 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+extension Metadata {
+    /// A directive that allows you to set a page's role and title heading.
+    ///
+    /// The `@PageKind` directive allows you to override a page's title heading to something other
```

Comment:
This also affects auto-curation behavior right? Is there a more general way we could describe this since the page kind affects more than just the title heading?

Reply:
To be honest i'm not sure if this affects auto-curation! Because of where the information is applied (`RenderNodeTranslator`) i think it's too late to mess with curation. (I would need to double-check, though.)

Reply:
I would say it _should_. If it doesn't now – that's fine, but I would view it as a bug we should fix.

If the directive is just about affecting the eyebrow then we should name it `@TitleHeadingKind` – not `@PageKind`.

Reply:
It looks like it doesn't affect automatic curation because the `AutomaticCuration` behavior is specific to symbols - automatic curation of articles only happens at the documentation root, and gets a hard-coded "Articles" heading. If this needs to be updated, it will likely require updating this code in `DocumentationContext` to create multiple task groups: https://github.com/apple/swift-docc/blob/9c28bc4391e04bc6eaf4edf6226e1b94ee93805e/Sources/SwiftDocC/Infrastructure/DocumentationContext.swift#L1881

Should that go into this PR?

This PR currently only processes `@PageKind` in articles, not symbols. I think if we allowed this in symbol extensions, it would affect curation a lot more, by moving symbol pages into the "Articles" task group instead of their symbol-kind task group. I'm not sure offhand if that would break code, since there would be a `Symbol` semantic type with an `article` page kind, which feels like it would break some assumptions in this code.

Reply:
Okay that's fine! If we can make the documentation clear that this is about changing the page's kind – and the kind happens to drive a page's default title heading and icon I think that would better. We'll likely have different directives for setting the page's title heading so it would be good to not conflate things here.

Reply:
I just pushed up some new docs for `PageKind`.